### PR TITLE
Feature/add typescript support

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: runner.js [hunk 1]
```diff
@@ -15,13 +15,37 @@ const buildArgs = (config, name, file) => {
...
+
+  installCmd && cmd.push(installCmd)
+  compileCmd && cmd.push(`${compileCmd} ${file}`)
+  runCmd && cmd.push(`${runCmd} ${file}`)
+
+  return cmd.join(' && ')
+}
+
+const cleanUpFiles = (config, name) => {
+  const deletes = config.outputExtensions.map(ext => (
```
Comment (stripedpajamas) [inline runner.js]: can be `arr.map(ext => fs.unlink(name.ext))` bc unlink is already a promise

File: runner.js [hunk 2]
```diff
@@ -15,13 +15,37 @@ const buildArgs = (config, name, file) => {
...
+    new Promise((resolve, reject) => {
+      fs.unlink(`${name}.${ext}`)
+        .then(() => resolve())
+        .catch(err => reject(err))
+    })
+  ))
+
+  return Promise.resolve(deletes)
```
Comment (stripedpajamas) [inline runner.js]: should be Promise.all

File: languages.json [hunk 3]
```diff
@@ -1,12 +1,22 @@
...
   },
   "py": {
     "extension": "py",
+    "outputExtensions": ["py"],
     "image": "python:3.7-slim",
-    "cmd": "python"
+    "runCmd": "python"
+  },
+  "ts": {
+    "extension": "ts",
```
Comment (stripedpajamas) [inline languages.json]: let's change extension to language or id so it's not confused with outputExtensions

**Review by stripedpajamas — APPROVED** [review]

File: runner.js [hunk 4]
```diff
@@ -2,41 +2,34 @@ const fs = require('fs').promises
...
   return new Promise((resolve, reject) => {
     const name = ulid.ulid()
-    const fileName = `${name}.${config.extension}`
+    const fileExtension = compileExtension || executeExtension
+    const fileName = utils.buildFileName(name, fileExtension)
+
     fs.writeFile(fileName, code).then(() => {
       let output = Buffer.alloc(0)
-      const docker = cp.spawn('docker', buildArgs(config, name, fileName))
+      const docker = cp.spawn('docker', utils.buildArgs(config, name, fileName))
```
Comment (stripedpajamas) [inline runner.js]: approved but also I think you're passing an extra arg here

Comment (kevin-orlando) [reply runner.js:20]: forgot to update, thanks

**Pull request merged**